Базовая стилизация всех страниц

AlexG-23 opened the pull request:
---
:mortar_board: [Базовая стилизация всех страниц](https://up.htmlacademy.ru/htmlcss/33/user/590651/tasks/6)

:boom: https://htmlacademy-htmlcss.github.io/590651-device-33/5/

AlexG-23 committed:
fixing of graphics

- delete alts of images in categories on index.html
-renaming of categories images
- add prev-arrow button icon of slider on index.html

AlexG-23 committed:
adding styles

- fixed and added classes to few elements
- add fonts
- add style to all texts on both pages
- took 'delivery-link' into div for background-color setting
- hid the elements with class 'visually-hidden'
- add some images in styles

AlexG-23 committed:
fixing of pagination section

- delete general background
- add border to "download-more" button
- add background to pagination links zone

LoremLipsum commented on catalog.html:
```diff
@@ -201,7 +202,7 @@ <h3 class="product-card-title">
         <article class="product-card">
           <a class="product-card-link" href="#">
-            <img class="product-card-image" src="images/catalog-items/product-2.jpg" alt="Профессиональная селфи-палка." width="360" height="380">
+            <img class="product-card-image" src="images/catalog-items/product-2.jpg" width="360" height="380">
```
Не надо удалять `alt`. Альт можно оставлять пустым, но он должен быть в разметке. Проверь через валидатор, сразу увидишь ошибки. 
`alt=""`
Также, в остальных картинках каталога

LoremLipsum commented on index.html:
```diff
@@ -159,47 +160,47 @@ <h2 class="visually-hidden">Категории товаров</h2>
           <a class="categories-link" href="#">
             <div class="categories-image-holder">
-              <img src="images/categories/category-1.svg" alt="Виртуальная реальность." width="97" height="55">
+              <img src="images/categories/vr-icon.svg" alt="Виртуальная реальность." width="97" height="55">
```
Мой косяк, надо убрать эти иконки из разметки, они не контентные. Обсудили в учительской. Это просто декоративные иконки.
Оставь просто тектс внутри ссылки, всё остальное убери

LoremLipsum commented on index.html:
```diff
@@ -238,7 +239,9 @@ <h3>Кредит</h3>
     </section>
 
-    <a href="#">доставим редкий товар под заказ</a>
+    <div class="delivery-link-wrapper">
```
Можно назвать блок просто `delivery`, враппер дабавится позже на этапе сеток

LoremLipsum commented on index.html:
```diff
@@ -273,7 +276,7 @@ <h2 class="subscribe-title">Подпишитесь на рассылку</h2>
       <form action="https://echo.htmlacademy.ru" method="post">    
         <label class="subscribe-field">
-          <input class="subscribe-email-field" type="email" name="email-field" placeholder="Ваш e-mail">
+          <input class="subscribe-email-field" type="email" name="email" placeholder="Ваш e-mail">
```
Добавь спан с скрытым текстовым описанием поля. Плейсхолдера может быть не достаточно для доступности

LoremLipsum commented on styles/styles.css:
```diff
@@ -38,7 +38,315 @@ body {
+}
+
+.page-header a {
+    text-decoration: none;
```
Нет, так писать не рекомендуется. В хэдере могут добавиться ссылки с подчеркиванием. Пропиши это именно для классов с ссылками

LoremLipsum commented on styles/styles.css:
```diff
@@ -38,7 +38,315 @@ body {
+
+.page-header a {
+    text-decoration: none;
+    color: #000000;
```
Здесь то же самое. Цвет надо прописать конкретным ссылкам

LoremLipsum commented on styles/styles.css:
```diff
@@ -38,7 +38,315 @@ body {
+
+.search {
+    font-size: 16px;
+    line-height: 20px;
```
Это шрифтовые свойства для кого? Если для инпута, пишу инпуту. Если для кнопки, пиши кнопке

LoremLipsum commented on styles/styles.css:
```diff
@@ -38,7 +38,315 @@ body {
+.pagination-item,
+.page-footer li {
+    list-style: none;
+}
```
Не делай так никогда. Не объединяй классы по одному свойству. Как это читать, править? 
И `list-style-type` обнуляется у списков, а не у лишек. 
Распиши все по отдельности и убери буллеты у списков, заодно можешь сразу обнулить у них маргины и паддинги

LoremLipsum commented on styles/styles.css:
```diff
@@ -38,7 +38,315 @@ body {
+    font-family: "Raleway", sans-serif;
+    font-weight: 800;
+    line-height: 21px;
+    text-transform: uppercase;
```
Вот здесь правильно. Текстовые свойсва написаны конкретно для элемента. Ещё, как вариант, можно было написать их для лишки, а не для ссылки

LoremLipsum commented on styles/styles.css:
```diff
@@ -38,7 +38,315 @@ body {
+    font-family: initial;
+    font-size: 16px;
+    font-weight: 400;
+    line-height: 18px;
```
Да можно и так, написать для лишки. Текстовые свойства надо ставить как можно ближе. И всегда смотри сразу в браузере чт получается

LoremLipsum commented on styles/styles.css:
```diff
@@ -38,7 +38,315 @@ body {
+******************************************************************************/
+
+.visually-hidden {
+    display: none;
```
Здесь должны быть другие свойства https://up.htmlacademy.ru/htmlcss/33/book/1.13

LoremLipsum commented on styles/styles.css:
```diff
@@ -38,7 +38,315 @@ body {
+    background: none;
+    text-transform: uppercase;
+    color: #000000;
+    text-decoration: none;
```
То же самое, не объединяй классы по одинаковым свойствам. Вы будуте проходить БЭМ на втором уровне. И там стили разделяются на отдельные блоки. Хэдер отдельно, футер тоже отдельно.

LoremLipsum commented on styles/styles.css:
```diff
@@ -38,7 +38,315 @@ body {
+.details dd {
+    font-family: initial;
+    font-size: 36px;
+    line-height: initial;
```
Что за initial? Пропиши нормально, чтобы всё читалось

LoremLipsum commented on styles/styles.css:
```diff
@@ -38,7 +38,315 @@ body {
+.categories-list h3 {
+    font-family: "Raleway", sans-serif;
+    line-height: 24px;
+    color: initial;
```
Опять initional. Как мне догадаться какой здесь цвет? Не злоупотребляй initial.  Пиши его только для общих случаях, где нужно поставить какой-то дефолт. Нпример для кнопок, там шрифт по дефолту определяется браузером, поэтому надо прописать font-family: initial. И то только в том случае, если шрифт для кнопок по смыслу совпадает с наборным для боди. А так лучше кокнретно прописать семейство шрифта

LoremLipsum commented on styles/styles.css:
```diff
@@ -38,7 +38,315 @@ body {
+}
+
+.categories-list a {
+    text-decoration: none;
```
Правильно. Обнуление дефолтных стилей непосредственно у самого элемента, чтобы не задеть остальные

LoremLipsum commented on styles/styles.css:
```diff
@@ -38,7 +38,315 @@ body {
+}
+
+.service-item:first-of-type:after {
+    content: url('../images/services/delivery.svg');
```
картинку надо ставить через` background-image`, а не `content`

LoremLipsum commented on styles/styles.css:
```diff
@@ -38,7 +38,315 @@ body {
+}
+
+.service-item:nth-of-type(2)::after {
+    content: url('../images/services/guarantee.svg');
```
background-image https://up.htmlacademy.ru/htmlcss/33/module/3/item/5

AlexG-23 committed:
fixes

- split union styles
- moved categories images to pseudo-elements
- changed all "initial" to required values
- changed the class name of delivery-link wrapper
- returned empty alt values to images
- added hidden span for accessibility
- split styles for links in navigation
- moved "list-style: none" from lis to lists (list-style-type)
- changed a way to hide elements
- moved pseudo-elements's images from content to background

AlexG-23 committed:
add pseudo-element

- add pseudo-element after delivery link

AlexG-23 committed:
change pseudo-element's style

- change a styles of after pseudo element of delivery link

AlexG-23 committed:
change pseudo-element's style

change pseudo elements before and after delivery link from display: block to inline-block

AlexG-23 committed:
last changes

- add class to button in filters section
- change image of breadcrumbs arrow
- add icons to navigation
- add bullets to list of transport companies on index page
- add more some icons on the pages
